Description: use Debian gems instead of upstream's embedded code copies
 Upstream ships Sonic Pi for Win, OS X and Raspbian and distributes all
 gems needed by the ruby server within the app/server/vendor/ directory
 in the source tarball. Debian wants packagers not to use these embedded
 code copies, but to use Debian's packaged ruby gems instead. This
 patch removes the code that added the vendor/ directory to the ruby
 library path. Do note that many of the gems provided by upstream
 aren't actually used in Sonic Pi, see
 <URL: https://github.com/samaaron/sonic-pi/issues/928> for details.
Forwarded: no
Author: Hanno Zulla <kontakt@hanno.de>
Last-Update: 2016-01-22

Index: sonic-pi-3.1.0~repack~test-1/app/server/ruby/core.rb
===================================================================
--- sonic-pi-3.1.0~repack~test-1.orig/app/server/ruby/core.rb
+++ sonic-pi-3.1.0~repack~test-1/app/server/ruby/core.rb
@@ -20,14 +20,8 @@ raise "Sonic Pi requires Ruby 2+ to be i
 require 'rbconfig'
 ruby_api = RbConfig::CONFIG['ruby_version']
 
-
-## Ensure all libs in vendor directory are available
-Dir["#{File.expand_path("../vendor", __FILE__)}/*/lib/"].each do |vendor_lib|
-  $:.unshift vendor_lib
-end
-
 begin
-  require 'did_you_mean'
+  require 'did_you_mean' if RUBY_VERSION >= "2.3.0"
 rescue LoadError
   warn "Non-critical error: Could not load did_you_mean"
 end
